### PR TITLE
Use user strings file with event categories cog

### LINKED ISSUE #52: Move user facing strings from EventCategoriesCog into user_strings.toml
In part completion for https://github.com/FragSoc/esports-bot/issues/41

### PR DESCRIPTION
**Pull request by Laura7089**
Closes #52.

Not yet ready for release, posting here to gather comments about admin logs among other things.

### CODE REVIEW

File: src/esportsbot/cogs/EventCategoriesCog.py [hunk 1]
```diff
@@ -19,68 +19,67 @@
 class EventCategoriesCog(commands.Cog):
     def __init__(self, bot: "EsportsBot"):
         self.bot: "EsportsBot" = bot
-
+        self.STRINGS = bot.STRINGS["admin"]
```
Comment (Trimatix) [inline src/esportsbot/cogs/EventCategoriesCog.py:22]:
```suggestion
        self.STRINGS = bot.STRINGS["event_categories"]
```

File: src/esportsbot/cogs/EventCategoriesCog.py [hunk 2]
```diff
@@ -19,68 +19,67 @@
...
             if not eventChannel.permissions_for(ctx.guild.me).manage_permissions:
-                await ctx.send(":x: I don't have permission to edit the permissions in <#" + str(eventChannel.id) + ">!")
+                await ctx.send(self.STRINGS['no_channel_edit_perms'].format(channel_id=eventChannel.id))
             else:
                 sharedRole = ctx.guild.get_role(guildData["shared_role_id"])
                 if not eventChannel.overwrites_for(sharedRole).read_messages:
-                    await eventChannel.set_permissions(sharedRole, read_messages=True,
-                                                    reason=ctx.author.name + f" opened the {eventName} event via the {self.bot.command_prefix}open-event command")
-                    await ctx.send("✅ <#" + str(eventChannel.id) + "> is now visible to **" + sharedRole.name + "**!")
+                    reason = self.STRINGS['event_channel_perms_reason'].format(author=ctx.author.name, event_name=eventName, command_prefix=self.bot.command_prefix)
```
Comment (Trimatix) [inline src/esportsbot/cogs/EventCategoriesCog.py:56]:
```suggestion
                    reason = self.STRINGS['event_channel_open_reason'].format(author=ctx.author.name, event_name=eventName, command_prefix=self.bot.command_prefix)
```

File: src/esportsbot/cogs/EventCategoriesCog.py [hunk 3]
```diff
@@ -205,65 +205,66 @@ async def admin_cmd_set_event_role(self, ctx: Context, *, args: str):
...
                         db = db_gateway()
                         if db.get("event_categories", {"guild_id": ctx.guild.id, "event_name": eventName}):
-                            await ctx.message.reply(f":x: An event category with the name '{eventName.title()}' already exists!")
+                            await ctx.message.reply(self.STRINGS['event_exists'].format(event_name=eventName.title()))
                         else:
                             menu = self.bot.reactionMenus[int(menuIDStr)]
                             db.insert('event_categories', {"guild_id": ctx.guild.id, "event_name": eventName, "role_id": roleID, "signin_menu_id": menu.msg.id})
-                            await ctx.send(f"✅ Event category '{eventName.title()}' registered successfuly!")
+                            await ctx.send(self.STRINGS['success_event_category'].format(event_name=eventName.title()))
+nothing_to_do = "Nothing to do!\n*(<#{channel_id!s}> already invisible to {shared_role}, no reactions on signin menu, no users with {event_role} role)*"
```
Comment (Trimatix) [inline src/esportsbot/cogs/EventCategoriesCog.py:233]:
```suggestion
```

File: src/esportsbot/cogs/EventCategoriesCog.py [hunk 4]
```diff
@@ -205,65 +205,66 @@ async def admin_cmd_set_event_role(self, ctx: Context, *, args: str):
...
                             signinEmoji = lib.emotes.Emote.fromPartial((await self.bot.wait_for("raw_reaction_add", check=emojiSelectorCheck, timeout=60)).emoji, rejectInvalid=True)
                         except asyncio.TimeoutError:
-                            await emojiSelectorMsg.reply("Out of time, please try the command again.")
+                            await emojiSelectorMsg.reply(self.STRINGS['react_no_time'])
                         except lib.exceptions.UnrecognisedCustomEmoji:
-                            await emojiSelectorMsg.reply("An error occurred when loading your reaction, please try the command again with a different emoji.")
+                            await emojiSelectorMsg.reply(self.STRINGS['react_error'])
                         else:
-                            creationReason = f"Creating new event category '{eventName}' requested via {self.bot.command_prefix}create-event-category command"
+                            creationReason = self.STRINGS['event_category_close_reason'].format(event_name=eventName, command_prefix=self.bot.command_prefix)
```
Comment (Trimatix) [inline src/esportsbot/cogs/EventCategoriesCog.py:267]:
```suggestion
                            creationReason = self.STRINGS['event_category_create_reason'].format(event_name=eventName, command_prefix=self.bot.command_prefix)
```

**Review by Trimatix — CHANGES_REQUESTED** [review]
*wow* you werent kidding, that was a lot of strings, thank you 😂
just a couple of typos to fix in there and then its ready to go!

**Comment by Laura7089** [comment]
Fixed all the above, there's a lot to keep track of with all the strings so mistakes slipped through!

I haven't added the admin logs yet, and intend to. Also, the logic for generating the loading text is a little complex and heavyweight for inserting into the toml file - can that be simplified with some clever format tricks to anyone's knowledge? If not, we can just split them up and build them in the code.

**Comment by Laura7089** [comment]
There, admin messages added, with a missed message. The implementation isn't very pretty, but it will do for now, I hope.

In the future, we could consider grouping the related messages into submaps within the `toml` file and then pulling them in in cog constructors separately.

**Review by Fluxticks — APPROVED** [review]

**Comment by Trimatix** [comment]
> Fixed all the above, there's a lot to keep track of with all the strings so mistakes slipped through!
> 
> I haven't added the admin logs yet, and intend to. Also, the logic for generating the loading text is a little complex and heavyweight for inserting into the toml file - can that be simplified with some clever format tricks to anyone's knowledge? If not, we can just split them up and build them in the code.
> 
> Edit: do we also want the command help messages to go into the file? They're user-facing messages and do match up with the error messages that are already in there...

love this idea but user strings would have to be accessible at module import time - right now they're only accessible at cog instance time

**Review by Trimatix — APPROVED** [review]

**Comment by Laura7089** [comment]
> love this idea but user strings would have to be accessible at module import time - right now they're only accessible at cog instance time

Gotcha. I thought that might be the case, which is why I didn't replace them. That sucks - I'll open a tracking issue which we might get around to eventually.

**Pull request merged**